code/modules/mob/living/simple_animal/revenant/revenant.dm

```diff
@@ -216,7 +216,7 @@
 	..()
 
 /datum/objective/revenant/check_completion()
-	if(!istype(owner.current, /mob/living/simple_animal/revenant) || !owner.current)
+	if(owner && !istype(owner.current, /mob/living/simple_animal/revenant) || !owner || !owner && !owner.current)
```

Comment:
To be more specific, short-circuiting is a feature of the binary logic operators themselves - they "return" the earliest value they can. In this case, since `!owner` is true, the `||` returns that without evaluating further. In the case of `&&`, it will return the first value if it's false (ignoring the next value), and return the second value if the first is true.

The logic operators can actually be used just about anywhere. `var/a = 1 || 2` will set `a` to 1, `var/b = 1 && 2` will set `b` to 2. Short-circuiting is pretty handy for brevity, but sometimes hard to read, so most people avoid using it "cleverly".